ZEngine/src/Helper.cpp

```diff
@@ -0,0 +1,879 @@
+        specification.RasterizationStateCreateInfo.lineWidth               = 1.0f;
+        specification.RasterizationStateCreateInfo.cullMode                = VK_CULL_MODE_BACK_BIT;
+        specification.RasterizationStateCreateInfo.frontFace               = VK_FRONT_FACE_CLOCKWISE;
+        specification.RasterizationStateCreateInfo.depthBiasEnable         = VK_FALSE;
+        specification.RasterizationStateCreateInfo.depthBiasConstantFactor = 0.0f; // Optional
+        specification.RasterizationStateCreateInfo.depthBiasClamp          = 0.0f; // Optional
+        specification.RasterizationStateCreateInfo.depthBiasSlopeFactor    = 0.0f; // Optional
+        specification.RasterizationStateCreateInfo.pNext                   = nullptr;
+
+        /*Multisampling*/
+        specification.MultisampleStateCreateInfo.sType                 = VK_STRUCTURE_TYPE_PIPELINE_MULTISAMPLE_STATE_CREATE_INFO;
+        specification.MultisampleStateCreateInfo.sampleShadingEnable   = VK_FALSE;
+        specification.MultisampleStateCreateInfo.rasterizationSamples  = VK_SAMPLE_COUNT_1_BIT;
+        specification.MultisampleStateCreateInfo.minSampleShading      = 1.0f;     // Optional
+        specification.MultisampleStateCreateInfo.pSampleMask           = nullptr;  // Optional
+        specification.MultisampleStateCreateInfo.alphaToCoverageEnable = VK_FALSE; // Optional
+        specification.MultisampleStateCreateInfo.alphaToOneEnable      = VK_FALSE; // Optional
+        specification.MultisampleStateCreateInfo.flags                 = 0;
+        specification.MultisampleStateCreateInfo.pNext                 = nullptr;
+
+        /*Depth and Stencil testing*/
+        specification.DepthStencilStateCreateInfo.sType                 = VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO;
+        specification.DepthStencilStateCreateInfo.depthTestEnable       = VK_TRUE;
+        specification.DepthStencilStateCreateInfo.depthWriteEnable      = VK_TRUE;
+        specification.DepthStencilStateCreateInfo.depthCompareOp        = VK_COMPARE_OP_LESS;
+        specification.DepthStencilStateCreateInfo.depthBoundsTestEnable = VK_FALSE;
+        specification.DepthStencilStateCreateInfo.minDepthBounds        = 0.0f; // Optional
+        specification.DepthStencilStateCreateInfo.maxDepthBounds        = 1.0f; // Optional
+        specification.DepthStencilStateCreateInfo.stencilTestEnable     = VK_FALSE;
+        specification.DepthStencilStateCreateInfo.front                 = {}; // Optional
+        specification.DepthStencilStateCreateInfo.back                  = {}; // Optiona
+
+        /*Color blending*/
+        // This configuration is per-framebuffer definition
+        specification.ColorBlendAttachmentState.colorWriteMask      = VK_COLOR_COMPONENT_R_BIT | VK_COLOR_COMPONENT_G_BIT | VK_COLOR_COMPONENT_B_BIT | VK_COLOR_COMPONENT_A_BIT;
+        specification.ColorBlendAttachmentState.blendEnable         = VK_TRUE;
```

Comment:
I would consider blending disabled to be a better default state, since blending isn't that cheap and having it enabled always is something that can be easily missed and potentially only seen in a capture. For example, with the default blending factors here, there will be virtually no difference in output if your fragment shader always writes alpha == 1.0.

Reply:
Fixed :) ... Thanks